ls1intum/Artemis
# `Discussion:` Introduce new post filter in course overview

### Checklist
#### General


#### Client

### Motivation and Context


A user (student or tutor) is especially interested in own posts and posts they answered to or reacted on. Those posts should be find quickly by the users, i.e. through activating filter in the course overview that match those characteristics.

### Description

Introduce two new filters (checkboxes):
- `Own`-> only show posts where current user is the author
- `Reacted or replied by you`-> only show posts where current user reacted or replied
Changed sort direction selection from dropdown text to icon -> there is not too much space with the new filters, I added tooltips, I thinks it's clear what the arrows are for and how to use them

These filters can be combined with the text search and the context selector and filter the loaded posts on the client side.

### Steps for Testing

Prerequisites:
- 1 Instructor / Student
- 1 Course with posts enabled -> e.g. https://artemistest.ase.in.tum.de/courses/477/discussion
- Different types of posts, resolved, answered, unanswered, with reactions.. 

1. Log in to Artemis
2. Navigate to course discussion
3. Use the filters (also in combination) -> check that the result matches your expectation

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





course-discussion.component.ts : 90%

### Screenshots


Filters in action:

https://user-images.githubusercontent.com/41366522/137397176-5557dc99-6a87-4a6c-a7b4-82c011b0f78f.mov


Filter/Sort Bar before:
<img width="1246" alt="Bildschirmfoto 2021-10-14 um 23 19 39" src="https://user-images.githubusercontent.com/41366522/137397158-412a051f-1322-499e-a7b4-6c95d20ecd71.png">


Filter/Sort Bar after:
<img width="1241" alt="Bildschirmfoto 2021-10-14 um 23 17 50" src="https://user-images.githubusercontent.com/41366522/137396938-ed10392d-0433-42d8-8578-c3cd7dde402d.png">
